src/libsystemd/sd-netlink/netlink-types-rtnl.c

```diff
@@ -902,13 +904,13 @@ DEFINE_TYPE_SYSTEM(rtnl_link);
 /* IFA_FLAGS was defined in kernel 3.14, but we still support older
  * kernels where IFA_MAX is lower. */
 static const NLType rtnl_address_types[] = {
-        [IFA_ADDRESS]           = { .type = NETLINK_TYPE_IN_ADDR },
-        [IFA_LOCAL]             = { .type = NETLINK_TYPE_IN_ADDR },
+        [IFA_ADDRESS]           = { .type = NETLINK_TYPE_IN_ADDR, .size = sizeof(struct in_addr) },
+        [IFA_LOCAL]             = { .type = NETLINK_TYPE_IN_ADDR, .size = sizeof(struct in_addr) },
         [IFA_LABEL]             = { .type = NETLINK_TYPE_STRING, .size = IFNAMSIZ - 1 },
-        [IFA_BROADCAST]         = { .type = NETLINK_TYPE_IN_ADDR },
-        [IFA_ANYCAST]           = { .type = NETLINK_TYPE_IN_ADDR },
+        [IFA_BROADCAST]         = { .type = NETLINK_TYPE_IN_ADDR, .size = sizeof(struct in_addr) },
+        [IFA_ANYCAST]           = { .type = NETLINK_TYPE_IN_ADDR, .size = sizeof(struct in_addr) },
```

Comment:
IFA_ANYCAST takes only IPv6. Not IPv4.

Reply:
Dropped IFA_BROADCAST, but used  in6_addr for IFA_ANYCAST.